build/kconfig/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+/// Application configuration passed into the kernel build.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct KernelConfig {
+    /// Tasks in the app image. The order of tasks is significant.
+    pub tasks: Vec<TaskConfig>,
+
+    /// Regions that tasks have shared access to, keyed by the name the task
+    /// config used to grant access (often peripheral name). These are typically
+    /// memory mapped peripherals (and should probably not be anything else).
```

Comment:
Two notable exceptions to "should probably not be anything else" that may be worth calling out: `bank2` and `system_flash`

Reply:
TBH neither of those should be represented using shared peripheral regions, we sort of need a new mechanism -- but in terms of this file, you are correct.